olp-cpp-sdk-authentication/include/olp/authentication/IntrospectAppResult.h

```diff
@@ -0,0 +1,235 @@
+   * @return
+   */
+  int GetTokenDuration() const;
+
+  /**
+   * @brief List Referrer URLs to register/registered with Application.
+   *
+   * On create/update: if the parameter is not present, the current value(s)
+   * are unchanged. If empty list specified, current value(s) will be reset to
+   * empty list.
+   * Value has a min length of 1 char and max of 255 chars.
+   * Wildcards are NOT allowed. The only valid characters: alphanumerics, '-',
+   * '_', '.', '/' The protocol is NOT specified, i.e. no http:// or https://
+   * Some examples: here.com, localhost, 127.0.0.1,
+   * www.facebook.com/hello/world/.
+   *
+   * @return The list Referrer URLs to register/registered with Application.
+   */
+  const std::vector<std::string>& GetReferrers() const;
+
+  /**
+   * @brief Status of the application.
+   *
+   * @return The status of the application.
+   */
+  const std::string& GetStatus() const;
+
+  /**
+   * @brief Identifies if application codes are enabled.
+   *
+   * @return True if application codes are enabled and false otherwise.
+   */
+  bool GetAppCodeEnabled() const;
+
+  /**
+   * @brief Timestamp (milliseconds since the Unix epoch) of when
+   * the application was created.
+   *
+   * @return Epoch time when the application was created.
+   */
+  time_t GetCreatedTime() const;
+
+  /**
+   * @brief Realm the application belongs to.
+   *
+   * @return The string containing application realm.
+   */
+  const std::string& GetRealm() const;
+
+  /**
+   * @brief Type of the application.
+   *
+   * @return The type of the application.
+   */
+  const std::string& GetType() const;
+
+  /**
+   * @brief List of response types.
+   *
+   * @return The list of response types.
+   */
+  const std::vector<std::string>& GetResponseTypes() const;
+
+  /**
+   * @brief Rate limit tier to configure application for.
+   *
+   * @return The string containing application rate limit tier.
+   */
+  const std::string& GetTier() const;
+
+  /**
+   * @brief HRN of the application.
+   *
+   * @return The string with application HRN.
+   */
+  const std::string& GetHRN() const;
+
+ private:
+  friend class AuthenticationClient;
+
+  void SetClientId(std::string client_id);
+  void SetName(std::string name);
+  void SetDescription(std::string description);
+  void SetReditectUris(std::vector<std::string> uris);
+  void SetAllowedScopes(std::vector<std::string> scopes);
+  void SetTokenEndpointAuthMethod(std::string method);
+  void SetTokenEndpointAuthMethodReason(std::string reason);
+  void SetDobRequired(bool is_required);
+  void SetTokenDuration(int duration);
+  void SetReferrers(std::vector<std::string> urls);
+  void SetStatus(std::string status);
+  void SetAppCodeEnabled(bool is_enabled);
+  void SetCreatedTime(time_t created_time);
+  void SetRealm(std::string realm);
+  void SetType(std::string type);
+  void SetResponseTypes(std::vector<std::string> types);
+  void SetTier(std::string tier);
+  void SetHrn(std::string hrn);
```

Comment:
You don't need to make them private. They can be public. It is a user problem if he overrides the values.